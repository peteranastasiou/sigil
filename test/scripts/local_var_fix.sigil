
# Test cases for fixing how local variables work.
# The assumption that all values left on the stack are locals (after each line) is not true for the grammar

var f = fn(){ print("a"+{ var b = "b"; var c = "c"; b+c }); };
f();

var g = fn(){ print(for i in 0:3 {""+i} + for i in 6:9 {""+i}); };
g();

echo ""+for i in 1:5 {2*i};

var h = fn(){ echo ""+for i in 1:5 {2*i}; };
h();
