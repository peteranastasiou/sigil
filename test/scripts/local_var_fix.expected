abc
[0,1,2,6,7,8]
[2,4,6,8]
[2,4,6,8]
